drivers/ipm/ipm_mhu.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <soc.h>
+#include "ipm_mhu.h"
+
+#define DEV_CFG(dev) \
+	((const struct ipm_mhu_device_config * const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct ipm_mhu_data *)(dev)->driver_data)
+#define IPM_MHU_REGS(dev) \
+	((volatile struct ipm_mhu_reg_map_t  *)(DEV_CFG(dev))->base)
+
+#if defined(CONFIG_IPM_MHU0) || defined(CONFIG_IPM_MHU1)
```

Comment:
This ifdef shouldn't be needed.

Reply:
CONFIG_IPM=y
CONFIG_IPM_MHU=y
#CONFIG_IPM_MHU0=y
#CONFIG_IPM_MHU1=y

If config file above, there are warnings about some symbols declared but not used. Do we need to consider this scenario?